File: app/models/candidate.rb
# frozen_string_literal: true

class Candidate < ApplicationRecord
  belongs_to :evaluator, class_name: 'User', optional: true

  has_many :recruitments, dependent: :restrict_with_error

  # # Validations
  #
  validates :email, presence: true, uniqueness: true, format: URI::MailTo::EMAIL_REGEXP
  validates :first_name, :last_name, presence: true

  # # Methods
  #
  def fullname
    [first_name, last_name].join(' ').strip
  end
end

File: app/javascript/models/candidate.js
import { Model, List } from './base'
import i18n from '@locales/i18n'

class Candidate extends Model {
  get defaults() {
    return {
      id: null,
      first_name: '',
      last_name: '',
      email: '',
      gender: ''
    };
  }

  get fullname() {
    return [this.first_name, this.last_name].join(' ')
  }

  static get routes() {
    return {
      candidatesPath: '/v2/candidates',
      candidatePath: id => `/v2/candidates/${id}`
    }
  }
}

class CandidatesList extends List {
  get model() {
    return Candidates;
  }
}

export { Candidate, CandidatesList };
